### PR TITLE
Feat/forgot password

### PR DESCRIPTION
**Pull request by micbaumr**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Screenshot(s)

Please attach some screenshot or a movie if provided changes affects UI.

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [x] Add correct label to your pull request
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/wrappers/FormWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { useScreenOptions } from '@baca/hooks'
+import { FC, PropsWithChildren } from 'react'
+import { StyleSheet } from 'react-native'
+
+import { KeyboardAwareScrollView } from '../KeyboardAwareScrollView'
+
+export const FormWrapper: FC<PropsWithChildren> = ({ children }) => {
+  useScreenOptions({ headerShown: false })
```
Comment (MateuszRostkowski) [inline src/components/wrappers/FormWrapper.tsx:8]: **Suggestion:** 
Maybe this option should come with props 🤔 

It's very common that we will don't want to hide header.

File: src/navigation/tabNavigator/components/AppHeader.tsx [hunk 2]
```diff
@@ -38,6 +25,6 @@ const jsStyles = StyleSheet.create({
     paddingHorizontal: 16,
     width: '100%',
     zIndex: 10,
+    ...Platform.select({ default: { display: 'none' }, web: { display: 'flex' } }),
```
Comment (MateuszRostkowski) [inline src/navigation/tabNavigator/components/AppHeader.tsx:28]: Maybe it's better to not render this component on all platforms:

```
if (Platform.OS !== 'web') {
  return null
}
```

Thanks to that we will not run all of this logic that is in return.

File: src/screens/auth/SignInScreen.tsx [hunk 3]
```diff
@@ -31,20 +34,20 @@ export const SignInScreen = (): JSX.Element => {
...
   const navigateToSignUp = useCallback(() => push('/sign-up'), [push])
   const navigateToAppInfo = useCallback(() => push('/application-info'), [push])
+  const navigateToForgotPassword = useCallback(
+    () => push(`/forgot-password?email=${encodeURIComponent(getValues('email'))}`),
+    [getValues, push]
+  )
   const focusPasswordInput = useCallback(() => setFocus('password'), [setFocus])
 
   return (
     <KeyboardAwareScrollView>
```
Comment (MateuszRostkowski) [inline src/screens/auth/SignInScreen.tsx:44]: You could add FormWrapper here

File: src/screens/auth/SignUpScreen.tsx [hunk 4]
```diff
@@ -25,6 +25,9 @@ export const SignUpScreen = () => {
   return (
     <KeyboardAwareScrollView>
```
Comment (MateuszRostkowski) [inline src/screens/auth/SignUpScreen.tsx:26]: You could add FormWrapper here.

**Review by MateuszRostkowski — APPROVED** [review]
Looks good! I've added some suggestions, but they are not needed.

File: src/screens/auth/SignInScreen.tsx (hunk 3 above)
Comment (micbaumr) [reply src/screens/auth/SignInScreen.tsx]: Added

File: src/screens/auth/SignUpScreen.tsx (hunk 4 above)
Comment (micbaumr) [reply src/screens/auth/SignUpScreen.tsx]: Added

**Pull request merged**